geode-core/src/main/java/org/apache/geode/cache/InterestPolicy.java

```diff
@@ -104,9 +100,9 @@ private Object readResolve() throws ObjectStreamException {
 
 
   /** Creates a new instance of InterestPolicy. */
-  private InterestPolicy(String name) {
+  private InterestPolicy(String name, int ordinal) {
     this.name = name;
-    this.ordinal = nextOrdinal++;
+    this.ordinal = (byte) ordinal;
```

Comment:
Would it be more appropriate to treat ordinal as either a byte or int universally wherever it is used instead of downcasting?

Reply:
One annoying thing about java is that there is no way to declare a byte literal. So we can either cast here in the constructor, or when we declare the constants - like this

```
public static final InterestPolicy CACHE_CONTENT = new InterestPolicy("CACHE_CONTENT", *(byte)* 1);
```

If you prefer the above approach, I can change it.

Reply:
Oh ok - no I think between the two I am with you that casting in the constructor is a little more explicit.  Thanks for clarifying!